compat/src/react-18-hooks.js

```diff
@@ -0,0 +1,54 @@
+import { useState, useLayoutEffect, useEffect } from 'preact/hooks';
+
+/**
+ * Asynchronously schedule a callback
+ * @type {(cb: () => void) => void}
+ */
+/* istanbul ignore next */
+// Note the following line isn't tree-shaken by rollup cuz of rollup/rollup#2566
+const defer =
+	typeof Promise == 'function'
+		? Promise.prototype.then.bind(Promise.resolve())
+		: setTimeout;
+
+export function useDeferredValue(val) {
+	return val;
+}
+
+export function useTransition() {
+	const [isPending, setIsPending] = useState(false);
+
+	const start = cb => {
```

Comment:
I believe this function should be the same on very render, so `useCallback` should be used.